### PR TITLE
[Dev][Bugfix] Add RMS Normalization Kernels and Fix Reduce Bug

### PR DESCRIPTION
**Pull request by chengyupku**
- Implement two RMS normalization implementations in TileLang:
  * `rms_norm_splitk`: Split-K reduction approach for large matrices
  * `rms_norm`: Full reduction kernel with simplified implementation
- Add reference implementation using PyTorch for validation
- Include performance benchmarking for both kernel variants
- Demonstrate flexible block size and matrix size configurations

### CODE REVIEW

**Review by LeiWang1999 — COMMENTED** [review]
We should add license to this file.

File: examples/norm/rms_norm.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import torch
```
Comment (LeiWang1999) [inline examples/norm/rms_norm.py:4]: license required.

File: examples/norm/rms_norm.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+if __name__ == "__main__":
+    M, N, blk_m, blk_k = 8192, 8192, 1, 512
+    # program = rms_norm_splitk(M, N, blk_m, blk_k)
+    program = rms_norm(M, N, blk_m)
+    kernel = tilelang.compile(
+        program,
+        out_idx=-1,
+        target="cuda",
+        execution_backend="cython",
+        pass_configs={"tl.disable_tma_lower": True})
```
Comment (LeiWang1999) [inline examples/norm/rms_norm.py]: Why we need `"tl.disable_tma_lower"` ?

File: examples/norm/rms_norm.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    return main
+
+
+def ref_program(x):
+    return x * torch.rsqrt(x.pow(2).mean(-1, keepdim=True) + 1e-12)
+
+
+if __name__ == "__main__":
+    M, N, blk_m, blk_k = 8192, 8192, 1, 512
+    # program = rms_norm_splitk(M, N, blk_m, blk_k)
```
Comment (LeiWang1999) [inline examples/norm/rms_norm.py]: comments should be removed :)

Comment (chengyupku) [reply examples/norm/rms_norm.py]: OK, done.

File: examples/norm/rms_norm.py (hunk 2 above)
Comment (chengyupku) [reply examples/norm/rms_norm.py]: Looks unnecessary, removed.

File: examples/norm/rms_norm.py (hunk 1 above)
Comment (chengyupku) [reply examples/norm/rms_norm.py:4]: done
